.github/workflows/kind-e2e.yaml

```diff
@@ -203,6 +208,59 @@ jobs:
         echo "GATEWAY_OVERRIDE=$GATEWAY_OVERRIDE" >> $GITHUB_ENV
         echo "GATEWAY_NAMESPACE_OVERRIDE=$GATEWAY_NAMESPACE_OVERRIDE" >> $GITHUB_ENV
 
+    - name: Apply NetworkPolicy
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -x
+
+        cat >> netpolicy.yaml <<EOF
+        apiVersion: networking.k8s.io/v1
+        kind: NetworkPolicy
+        metadata:
+          name: default-deny
+          namespace: $SYSTEM_NAMESPACE
+        spec:
+          podSelector:
+            matchExpressions:
+            - key: app
+              operator: In
+              values: [autoscaler, autoscaler-hpa]
+          policyTypes:
+          - Ingress
+          ingress:
+          - from:
+            - namespaceSelector:
+                matchLabels:
+                  # kubernetes.io/metadata.name: $SYSTEM_NAMESPACE # this only works on k8s >= 1.21.1
```

Comment:
Is the idea of keeping this line here commented that we should eventually use this in preference to the other label below? If so I'd suggest making the comment something like "TODO: use this in preference to the projectcalico.org/name label after minVersion > k8s 1.21.1" so we remember to do this when we can